src/components/form-spinbutton/form-spinbutton.js

```diff
@@ -157,35 +145,35 @@ export const BFormSpinbutton = /*#__PURE__*/ Vue.extend({
   },
   data() {
     return {
-      localValue: defaultNumber(this.value),
+      localValue: toFloat(this.value, null),
       hasFocus: false
     }
   },
   computed: {
     computedStep() {
-      return defaultNumber(this.step, DEFAULT_STEP)
+      return toFloat(this.step, DEFAULT_STEP)
     },
     computedMin() {
-      return defaultNumber(this.min, DEFAULT_MIN)
+      return toFloat(this.min, DEFAULT_MIN)
     },
     computedMax() {
       // We round down to the nearest maximum step value
-      const max = defaultNumber(this.max, DEFAULT_MAX)
+      const max = toFloat(this.max, DEFAULT_MAX)
       const step = this.computedStep
       const min = this.computedMin
       return Math.floor((max - min) / step) * step + min
     },
     computedDelay() {
-      return defaultInteger(this.repeatDelay, DEFAULT_REPEAT_DELAY) || DEFAULT_REPEAT_DELAY
+      return toInteger(this.repeatDelay, DEFAULT_REPEAT_DELAY)
```

Comment:
Should be:

```
retutn toInteger(this.repeatDelay, DEFAULT_REPEAT_DELAY) || DEFAULT_REPEAT_DELAY
```

As we don't want a `0` value here if the user provides `0`

Reply:
`return toInteger(this.repeatDelay, 0) || DEFAULT_REPEAT_DELAY` should work too then and makes it a bit cleared what we cant to archive.

Reply:
Yeah